Server/VikingVault.Services/TransferFundsService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Text;
+using VikingVault.DataAccess;
+using VikingVault.DataAccess.Models;
+using VikingVault.DataAccess.Models.Exceptions;
+using VikingVault.Services.Abstractions;
+using VikingVault.Services.Exceptions.CardException;
+
+namespace VikingVault.Services
+{
+    public class TransferFundsService : ITransferFundsService
+    {
+        private readonly VikingVaultDbContext _dbContext;
+        private readonly IBankAccountService _bankAccountService;
+        private readonly ITransactionService _transactionService;
+        private readonly IUserCardService _userCardService;
+        private readonly IUserService _userService;
+
+        public TransferFundsService(VikingVaultDbContext dbContext, IBankAccountService bankAccountService, ITransactionService transactionService, IUserCardService userCardService, IUserService userService)
+        {
+            _dbContext = dbContext;
+            _bankAccountService = bankAccountService;
+            _transactionService = transactionService;
+            _userCardService = userCardService;
+            _userService = userService;
+        }
+
+        public bool? TransferFunds(TransferFundsModel transferData)
+        {
+            int idSender = transferData.Sender.Id;
```

Comment:
You have so many if`s in this method... You shall refactor it in order to use fewer if`s or at least, extract some of them to separate methods.